tests/lib/UTILS.groovy

```diff
@@ -1,11 +1,382 @@
-// Function to remove Nextflow version from pipeline_software_mqc_versions.yml
-
 class UTILS {
-    public static String removeNextflowVersion(pipeline_software_mqc_versions) {
+    // Remove Nextflow version from pipeline_software_mqc_versions.yml
+    public static Object removeNextflowVersion(pipeline_software_mqc_versions) {
         def softwareVersions = path(pipeline_software_mqc_versions).yaml
-        if (softwareVersions.containsKey("Workflow")) {
-            softwareVersions.Workflow.remove("Nextflow")
-        }
+        if (softwareVersions.containsKey("Workflow")) softwareVersions.Workflow.remove("Nextflow")
         return softwareVersions
     }
+
+    // Recursively list all files in a directory and its sub-directories, matching or not matching supplied suffixes
+    public static getAllFilesFromDir(dir, List<String> includeRegexes = null, List<String> excludeRegexes = null) {
+        def output = []
+        new File(dir).eachFileRecurse() { file ->
+            boolean matchesInclusion = includeRegexes ? includeRegexes.any { regex -> file.name.toString() ==~ regex } : true
+            boolean matchesExclusion = excludeRegexes ? excludeRegexes.any { regex -> file.name.toString() ==~ regex } : false
+
+            // Add files to the list if they match the includeRegexes and not the excludeRegexes
+            if (matchesInclusion && !matchesExclusion) output.add(file)
+        }
+        return output.sort { it.path }
+    }
```

Comment:
```suggestion
    public static getAllFilesFromDir(dir, List<String> includeRegexes = null, List<String> excludeRegexes = null, boolean relativePaths = false) {
        def output = []
        def dirFile = new File(dir)
        dirFile.eachFileRecurse { file ->
            boolean matchesInclusion = includeRegexes ? includeRegexes.any { regex -> file.name.toString() ==~ regex } : true
            boolean matchesExclusion = excludeRegexes ? excludeRegexes.any { regex -> file.name.toString() ==~ regex } : false

            // Add files to the list if they match the includeRegexes and not the excludeRegexes
            if (matchesInclusion && !matchesExclusion) {
                if (relativePaths) {
                    def relativePath = dirFile.toURI().relativize(file.toURI()).getPath()
                    output.add(new File(relativePath))
                } else {
                    output.add(file)
                }
            }
        }
        return output.sort { it.path }
    }
```

Let's allow this function to return relative paths, so the file complement snapshot is more understandable

Reply:
ok, I merged that, but we don't have relative paths in the snapshots